src/ui/src/shape_list_widget.cpp

```diff
@@ -1,10 +1,172 @@
 #include "shape_list_widget.h"
 
-#include "ui_shape_list_widget.h"
+#include <QStringList>
+#include <QFile>
+
 
 ShapeListWidget::ShapeListWidget(QWidget* parent)
-		: QDockWidget(parent), ui_(new Ui::ShapeListWidget) {
-	ui_->setupUi(this);
+		: QDockWidget(parent) {
+
+	// example with checkboxes https://stackoverflow.com/questions/14158191/qt-qtreeview-and-custom-model-with-checkbox-columns
+	
+	model_ = new TreeModel(this);
+	view_ = new QTreeView(this);
+	view_->setModel(model_);
+	view_->setWindowTitle(QObject::tr("Simple Tree Model"));
+	setWidget(view_);
+
+	Test();
+}
+
+void ShapeListWidget::Test() {
+	model_->TestFillWithTxtFile(
+			"C:\\Users\\winroot\\Desktop\\overheat\\local_docs\\default.txt");
+}
+
+// TreeItem
+
+
+TreeItem::TreeItem(const QVector<QVariant>& data, TreeItem* parent)
+		: m_itemData(data), m_parentItem(parent) {}
+
+TreeItem* TreeItem::child(int row) {
+	if (row < 0 || row >= m_childItems.size())
+		return nullptr;
+	return m_childItems.at(row);
+}
+
+QVariant TreeItem::data(int column) const {
+	if (column < 0 || column >= m_itemData.size())
+		return QVariant();
+	return m_itemData.at(column);
+}
+
+int TreeItem::row() const {
+	if (m_parentItem)
+		return m_parentItem->m_childItems.indexOf(const_cast<TreeItem*>(this));
+
+	return 0;
+}
+
+// Tree Model
+
+TreeModel::TreeModel(QObject* parent)
+		: QAbstractItemModel(parent) {
+	rootItem = new TreeItem({tr("Name"), tr("Visibility"), tr("Highlight")});
+}
+
+int TreeModel::columnCount(const QModelIndex& parent) const {
+	if (parent.isValid())
+		return static_cast<TreeItem*>(parent.internalPointer())->column_count();
+	return rootItem->column_count();
 }
 
-ShapeListWidget::~ShapeListWidget() { delete ui_; }
+QVariant TreeModel::data(const QModelIndex& index, int role) const {
+	if (!index.isValid())
+		return QVariant();
+	if (role != Qt::DisplayRole)
+		return QVariant();
+	TreeItem* item = static_cast<TreeItem*>(index.internalPointer());
+	return item->data(index.column());
+}
+
+Qt::ItemFlags TreeModel::flags(const QModelIndex& index) const {
+	return !index.isValid() ? QAbstractItemModel::flags(index) : Qt::NoItemFlags;
+}
+
+QVariant TreeModel::headerData(int section, Qt::Orientation orientation,
+                               int role) const {
+	if (orientation == Qt::Horizontal && role == Qt::DisplayRole)
+		return rootItem->data(section);
+	return QVariant();
+}
+
+QModelIndex TreeModel::index(int row, int column,
+                             const QModelIndex& parent) const {
+	if (!hasIndex(row, column, parent))
+		return QModelIndex();
+	TreeItem* parentItem;
```

Comment:
Please always initialize all variables. After refactorings it may be read uninitialized accidentially.
Maybe it make sense to init it here with `rootItem` and reassign with `internalPointer()` after check (saves a little perfomance theoretically)